editor/src/components/canvas/canvas-utils.ts

```diff
@@ -2424,7 +2424,7 @@ export function cullSpyCollector(
   })
   // Eliminate the element paths which are invalid.
   fastForEach(Object.keys(spyCollector.current.spyValues.metadata), (elementPath) => {
-    if (!elementPaths.has(elementPath)) {
+    if (!elementPaths.has(elementPath.split(':')[0])) {
```

Comment:
if we change this, then we also need to purge the metadata in the scene spy function! otherwise deleted elements will stick around